specfile/specfile.py

```diff
@@ -0,0 +1,106 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import contextlib
+import types
+from pathlib import Path
+from typing import Iterator, List, Optional, Tuple, Type, Union
+
+from specfile.rpm import RPM, Macros
+from specfile.sections import Sections
+from specfile.tags import Tags
+
+
+class Specfile:
+    """
+    Class representing spec file.
+
+    Attributes:
+        path: Path to the spec file.
+        sourcedir: Path to sources and patches.
+        autosave: Whether to automatically save any changes made.
+        macros: List of macro definitions.
+    """
+
+    def __init__(
+        self,
+        path: Union[Path, str],
+        sourcedir: Optional[Union[Path, str]] = None,
+        autosave: bool = False,
+        macros: Optional[List[Tuple[str, str]]] = None,
+    ) -> None:
+        self.path = Path(path)
+        self.sourcedir = Path(sourcedir or self.path.parent)
+        self.autosave = autosave
+        self.macros = macros.copy() if macros is not None else None
```

Comment:
Wouldn't an empty list `[]` be more convenient for further use than `None` if the argument is `None`?

Reply:
I don't think there will be any modification to the macros list before passing it to `RPM.parse()`, but sure, I'll change it to `[]` just in case.